pkg/apis/tenancy/v1alpha1/helper/helper.go

```diff
@@ -60,7 +60,7 @@ func EncodeOrganizationAndClusterWorkspace(organization, workspace string) strin
 // <OrganizationCluster>#$#<ws>. Otherwise, the key will be of the format
 // <OrganizationClsuter>_<org>#$#<ws>.
 func WorkspaceKey(org, ws string) string {
-	if org == RootCluster || strings.HasPrefix(org, LocalSystemClusterPrefix) {
+	if org == RootCluster || (org == "" && ws == RootCluster) || strings.HasPrefix(org, LocalSystemClusterPrefix) {
```

Comment:
is it only me who think this function is highly confusion?

I see calls with org being an org name, and calls where it is a org ClusterName.

Reply:
I definitely it our key handling confusing, and I think it is likely to be a persistent source of confusion and misunderstanding that all workspaces use the same type with very different behavior.